test/parser.jl

```diff
@@ -925,5 +925,8 @@ end""" |> test_expr
         x2 = CSTParser.minimal_reparse(s0, s1, x0, x1)
         @test CSTParser.comp(x1, x2)
     end
-    
+
+    @testset "end as id juxt" begin
+        @test test_expr("a[1end]")
```

Comment:
```suggestion
        @test test_expr("a[2begin:1end]")
```
so both special cases are covered?